examples/initiator/src/main/java/com/paritytrading/philadelphia/initiator/Initiator.java

```diff
@@ -105,22 +114,28 @@ FIXConnection getTransport() {
         return connection;
     }
 
-    void send(FIXMessage message) throws IOException {
-        sentAtNanoTime = System.nanoTime();
+    void reset() {
+        histogram.reset();
 
-        connection.update(message);
-        connection.send(message);
+        receiveCount = 0;
+    }
 
-        received = false;
+    void setIntervalNanos(long i) {
+        intervalNanos = i;
     }
 
-    void receive() throws IOException {
-        while (!received) {
-            if (connection.receive() < 0)
-                return;
-        }
+    void sendAndReceive(FIXMessage message, int orders) throws IOException {
+        for (long sentAtNanoTime = System.nanoTime(); receiveCount < orders; connection.receive()) {
+            if (System.nanoTime() >= sentAtNanoTime) {
+                FIXValue clOrdId = message.valueOf(ClOrdID);
```

Comment:
You could pass `FIXValue clOrdId` as a parameter to `sendAndReceive` to avoid invoking `FIXMessage#valueOf(int)` for each message: 🤓 
```java
void sendAndReceive(FIXMessage message, FIXValue clOrdId, int orders) throws IOException {
    ...
```